frontend/src/components/createUserModal/index.tsx

```diff
@@ -0,0 +1,194 @@
+import Checkbox from '@material-ui/core/Checkbox';
+import FormLabel from '@material-ui/core/FormLabel';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import Button from '@material-ui/core/Button';
+import Clear from '@material-ui/icons/Clear';
+
+import { createStaff } from '../../features/staffSlice';
+import { toDateStr } from '../../constants/global';
+
+import 'toastr/build/toastr.min.css';
+import createUserModalStyle from './createUserModal.module.scss';
+
+interface Props {
+  closeModalFunction: () => void;
+}
+
+interface TextField {
+  id: string;
+  name: string;
+  type: string;
+}
+
+const textFields: TextField[] = [
+  { id: 'firstName', name: 'First Name', type: 'text' },
+  { id: 'surname', name: 'Surname', type: 'text' },
+  { id: 'jobTitle', name: 'Job title', type: 'text' },
+  { id: 'contractedHours', name: 'Contracted hours', type: 'number' },
+  { id: 'pay', name: 'Pay', type: 'number' },
+];
+
+const CreateUserModal = ({ closeModalFunction }: Props) => {
+  const dispatch = useDispatch();
+  const [preferredDays, setPreferredDays] = useState<{
+    [key: string]: boolean;
+  }>({
+    Mon: false,
+    Tue: false,
+    Wed: false,
+    Thu: false,
+    Fri: false,
+    Sat: false,
+    Sun: false,
+  });
+
+  const [userInfo, setUserInfo] = useState<{ [key: string]: string | number }>({
+    firstName: '',
+    surname: '',
+    jobTitle: '',
+    role: 'USER',
+    contractedHours: 0,
+    pay: 0,
+  });
+
+  const handlePreferredDaysChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setPreferredDays({
+      ...preferredDays,
+      [event.target.name]: event.target.checked,
+    });
+  };
+
+  const updateUserInfo = (userField: string, newValue: string | number) => {
+    setUserInfo({
+      ...userInfo,
+      [userField]: newValue,
+    });
+  };
+
+  const handleTextFieldChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const userField = event.target.name as string;
+    const newValueString = event.target.value as string;
+    let newValue: string | number;
+
+    if (event.target.type === 'number') {
+      newValue = parseFloat(newValueString);
+    } else {
+      newValue = newValueString;
+    }
+
+    updateUserInfo(userField, newValue);
+  };
+
+  const handleDropdownChange = (
+    event: React.ChangeEvent<{ name?: string | undefined; value: unknown }>
+  ) => {
+    const userField = event.target.name as string;
+    const newValue = event.target.value as string | number;
+    updateUserInfo(userField, newValue);
+  };
+
+  const handleCreateClick = () => {
+    const prefDaysString = Object.values(preferredDays)
+      .map((preferred) => +preferred)
+      .join('');
+    dispatch(
+      createStaff({
+        firstName: userInfo.firstName as string,
+        surname: userInfo.surname as string,
+        jobTitle: userInfo.jobTitle as string,
+        role: userInfo.role as 'USER' | 'MANAGER',
```

Comment:
`'USER' | 'MANAGER'` is used a lot, I think this is a good use case for a string enum:
```
enum Role {
  USER = 'USER',
  MANAGER = 'MANAGER
}
```

Then use `Role` here, and replace `USER` with `Role.USER` and `'MANAGER'` with `Role.MANAGER`.

This comes with a few advantages:
- safety - in case of a typo, 'UESR' is a valid string, but not a valid role, so we want typescript to pick it up
- if we ever want to add another role, we'll just need to modify one enum.

Reply:
Good observation